pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -762,13 +762,27 @@ func (p *DNSProxy) CheckAllowed(endpointID uint64, destPort uint16, destID ident
 	return false, nil
 }
 
-func setSoMark(fd int, secId identity.NumericIdentity) error {
+func setSoMarks(fd int, ipFamily ipfamily.IPFamily, secId identity.NumericIdentity) error {
```

Comment:
Would it be possible to add comments in this function which would explain why we are setting these options? This might be out of scope of this PR, but I cannot figure out why we are setting reuseaddr here based on commit message/code .

Reply:
Added :-)